xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-xwiki/src/main/resources/ApplicationResources.properties

```diff
@@ -42,6 +42,12 @@
 #     old.key.name=Some translation
 ###############################################################################
 
+# Those entries are used to label the message boxes of different types.
```

Comment:
This text contradicts the generality of the icon provider. The icon provider should not just be usable for message boxes. Also, it is not clear to me why the icon provider is the one that is responsible for providing an alternative text. Wouldn't it make more sense that the code that calls the icon provider should also add the alternative text?

Reply:
I think this should be removed again now that the translations are handled by the overridden macros.

Reply:
The translation values are still needed in the overridden macros though. I kept them in, let me know if I misunderstood something here :)

Reply:
As far as I can see, they are now provided with identical keys in two different files. Is that really intentional and necessary?

Reply:
> As far as I can see, they are now provided with identical keys in two different files. Is that really intentional and necessary?

Same here, those looks like leftovers to me. The keys make more sense in the module where they are used (which is already the case AFAICS).